tests/suites/test_suite_pkcs7.data

```diff
@@ -0,0 +1,63 @@
+PKCS7 Signed Data Parse Pass SHA256 #1
+depends_on:MBEDTLS_PKCS7_C
+pkcs7_parse:"data_files/pkcs7_data_cert.signed.sha256"
+
```

Comment:
A number of test files are PEM-encoded, so the corresponding test cases need a dependency on `MBEDTLS_PEM_C`.

To check that the required dependencies are in place:
```
tests/scripts/all.sh -k test_no_pem_no_fs
```

But please minimize the use of PEM-encoded files, so that more unit tests can run in builds without `MBEDTLS_PEM_C`.

Reply:
> I do not call any PEM specific function directly, it is handled via x509_crt code. And I have already added dependency on X509_CRT_PARSE_C. For that reason, I am not sure if PEM dependency is really needed.

X.509 code optionally supports PEM files if `MBEDTLS_PEM_{PARSE,WRITE}_C` is defined. If you have PEM-encoded data, `MBEDTLS_PEM_PARSE_C` is needed in addition to the module(s) for what the data represents (`MBEDTLS_RSA_C` for an RSA key or an RSA-signed certificate, `MBEDTLS_X509_CRT_PARSE_C` for any certificate, etc.).

> I am having error running - tests/scripts/all.sh -k test_no_pem_no_fs, as it looks for this i686-w64-mingw32-gcc.

You don't need mingw to run `tests/scripts/all.sh -k test_no_pem_no_fs`. (`all.sh` displays a message about it being not found, but then it just decides that it can't do mingw builds and goes on with its life. If that's not the case for you, please file a bug with a full transcript of `tests/scripts/all.sh -k test_no_pem_no_fs`.)

Reply:
We have a [test gap around PEM](https://github.com/ARMmbed/mbedtls/issues/3915). So there may be preexisting missing dependencies on PEM. But please try to get them mostly right in new code.